src/xtopcom/xtopcl/src/api_method.cpp

```diff
@@ -685,8 +691,46 @@ void ApiMethod::query_miner_info(std::string & account, std::ostringstream & out
     string null_out = g_userinfo.account + " account has not registered miner.";
     tackle_null_query(out_str, null_out);
 }
+int ApiMethod::parse_top_double(const std::string &amount, const uint32_t unit, uint64_t &out)
```

Comment:
logic of this API can be simplified as:
1. looking for `.` and count it. if greater than 1, return; std::count(begin, end, '.')
2. check if all chars in the string are between '0' and '9' or is '.'. if not return; std::all_of(begin, end, [](...) { '0' ~ '9' or '.'});
3. trim leading '0' and tailing '0'
4. converting